### ISSUE TITLE
Allow to use DNS Service Discovery

### ISSUE THREAD

**Issue by ktaborski** [jcmoraisjr/haproxy-ingress#758]
**What are you trying to do**
I am using tcp services, where single connection can take several seconds. In case of HAProxy reloads, backend Pods can get overloaded (new HAProxy process does not know amount of previous connections). I found, that HAProxy has possibility to use DNS service discovery:
https://www.haproxy.com/blog/dns-service-discovery-haproxy/
Using this will allow to update/scale backend without HAProxy reloads.

**What HAProxy Ingress should do or how it should behave differently**
Probably I can still use it by defining my custom haproxy.cfg template, but in my opinion it would be useful to community if this will be configured in configMap (for both http and tcp)  and properly documented.

**Comment by jcmoraisjr** [comment]
Already in the v0.13 roadmap - we'll allow to configure tcp services using ingress resources, this way all L4 compatible backend configurations can be reused - including but not limited to dynamic updating - with or without dns discovery. You can follow #750 which will be included in the first v0.13 snapshot.

**Comment by ktaborski** [comment]
Thanks for your reply!

It's good, that this is under implementation.

After some digging I found, that it would be also good to allow usage of `parse-resolv-conf` function to define DNS resolvers
https://www.haproxy.com/documentation/hapee/2-1r1/configuration/config-sections/resolvers/

What do you think?

**Comment by jcmoraisjr** [comment]
Yep, `parse-resolv-conf` seems to be a good addition, at least as a config-snippet-like option.
